app/src/test/java/com/google/android/ground/ui/offlineareas/selector/OfflineAreaSelectorFragmentTest.kt

```diff
@@ -125,4 +132,31 @@ class OfflineAreaSelectorFragmentTest : BaseHiltTest() {
 
     viewModel.downloadProgress.removeObserver(observer)
   }
+
+  @Test
+  fun `test failure case displays toast`() = runWithTestDispatcher {
+    val isFailureObserver = mock(Observer::class.java) as Observer<Boolean>
+    fragment.viewLifecycleOwner.lifecycleScope.launch {
+      viewModel.isFailure.observeForever(isFailureObserver)
+      viewModel.isFailure.postValue(true)
```

Comment:
Is it possible to call the download flow and mock the `downloadTiles` to throw an error?

Reply:
let me try this in another pr